### PR TITLE
Refactor RecordServiceAdditionalResourceTypeTest

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This refactors and renames RecordServiceAdditionalResourceTypeTest.

## Motivation and Context

This depends on #61.
The currently RecordServiceAdditionalResourceTypeTest is a very bad shape - it is slow, hard to understand and applied only on Fhir3.

## How is it being implemented?
This makes a complete refactoring of the test set and labels it as a module test.
The tests are now run as a parameterized test, so it is more easier to add new resource types and let them run through all tests. Also the resources are not builded on runtime - they now loaded from the `resources` folder.
The attachment pdfs are also build only once, since they caused a massiv memory problem.
The tests are now running on Fhir4 resources as well.
Also it decouples the tests from the rest of the RecordServiceTest, which decreases the runtime of the tests by a lot.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by ouafae31 — DISMISSED** [review]
RecordServiceAdditionalResourceTypeTest LGTM.

File: sdk-core/src/test/java/care/data4life/sdk/RecordServiceAdditionalResourceTypeModule.kt [hunk 1]
```diff
@@ -942,8 +950,8 @@ class RecordServiceAdditionalResourceTypeModule {
                             SdkAttachmentFactory.wrap(firstAttachment),
                             SdkAttachmentFactory.wrap(secondAttachment)
                     ),
-                    any(),
-                    any()
+                    attachmentKey,
+                    USER_ID
             )
```
Comment (ouafae31) [inline sdk-core/src/test/java/care/data4life/sdk/RecordServiceAdditionalResourceTypeModule.kt:955]: Why is it important to remove any() ?

**Review by ouafae31 — APPROVED** [review]
I would prefer RecordServiceAdditionalResourceTypeModule to have test suffix to find it easily.
Good refactoring that allowed us to test all cases for all resourceType but unfortunately the readability of the each test with condition/consequence was reduced.

File: sdk-core/src/test/java/care/data4life/sdk/RecordServiceAdditionalResourceTypeModule.kt (hunk 1 above)
Comment (bitPogo) [reply sdk-core/src/test/java/care/data4life/sdk/RecordServiceAdditionalResourceTypeModule.kt:955]: `any()` in my opinion is never a good choice since you accept all payloads, since in a test environment the parameter should be known (except in some edge cases).  If you define the parameter specifically, the tests will not run and fail, which leads to much less problems with the test themselves.
Here I simply overlooked it the first time.

**Comment by bitPogo** [comment]
> I would prefer RecordServiceAdditionalResourceTypeModule to have test suffix to find it easily.
Good refactoring that allowed us to test all cases for all resourceType but unfortunately the readability of the each test with condition/consequence was reduced.

Can you elaborate a bit more, what you mean by that?

**Comment by ouafae31** [comment]
I just meant RecordServiceAdditionalResourceTypeModuleTest is better because I always expect '"test" suffix to differentiate between the class and its test.

**Review by ouafae31 — APPROVED** [review]
LGTM
